core/drivers/clk/clk.c

```diff
@@ -190,16 +223,23 @@ void clk_disable(struct clk *clk)
 
 unsigned long clk_get_rate(struct clk *clk)
 {
-	return clk->rate;
+	if (is_clk_elt(clk))
```

Comment:
Another comment for "drivers: clk: prepare for orphan clocks"
I'd rather use the `get_rate()` callback for this.

Reply:
true. make sense. thanks.

Reply:
soory. i"m wrong. `get_rate` is an internal call asking driver to return the effective clock based on platform clock state and a provided (possible) parent rate. we would need 2 operators, it would make how clk_elt are defined more complex: would mandate platform clock does register the generic handler, as done for `get_name`.

Reply:
The benefit would be a smaller code footprint for the orphan clocks.